src/runtime_src/core/tools/nagios/nagios_plugin.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2022 Xilinx, Inc
+# Copyright (C) 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+# Generate the output JSON file. Ignore both error and standard output
+/opt/xilinx/xrt/bin/xbutil examine $@ -f JSON-plain -o /tmp/nagios_output.json &> /dev/null
```

Comment:
Please refactor this line of code to do the following:
1. Update the script to accept a BDF value. 
    + If there are more then 1 device installed, xbutil will require that a device value be present.

2. Create a temporary unique output name for the JSON file.  
    + This will prevent name collisions for parallel processes.

3. Insert a simple python like script that will consume the output name (pretty print format) and write out a compressed JSON format.
    + This output would be added to the nagios log

4. Delete the temporary unique name.

An example on how this might be down can be seen with the the [xball](https://github.com/Xilinx/XRT/blob/master/src/runtime_src/core/tools/common/xball) script.

Reply:
The $@ option is replaced by all input arguments provided tot he nagios script. When running the nagios script I must invoke it via `./nagios_script -d 17:00 -r platform` (at least that is what I wrote in the documentation)
These changes are great! Thank you for the feedback and link to xball, it helped a lot.